### PR TITLE
Good 3 gas check

### PR DESCRIPTION
**Pull request by AlexeyKosinski**
https://github.com/GoodDollar/GoodDAPP/issues/435
adding fuse api to check if user has transactions with amount > 0 from user's wallet then user try to cheat us.

adding fuse api to check if user has transactions with value > 0 from user's wallet then user try to cheat us. so we don't need to refill his wallet with ether for make transactions

### CODE REVIEW

File: src/server/server.config.js [hunk 1]
```diff
@@ -265,6 +265,12 @@ const conf = convict({
     format: '*',
     env: 'ROLLBAR_TOKEN',
     default: undefined
+  },
+  fuse: {
+    doc: 'Main url for fuse api',
+    format: '*',
+    env: 'FUSE_API',
+    default: 'https://explorer.fuse.io'
```
Comment (sirpy) [inline src/server/server.config.js]: dont put a default and make it a required, other wise we will miss it

File: src/server/utils/fuseapi.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import request from 'request'
+import qs from 'qs'
+class FuseApi {
+  constructor() {
+    this.mainUrl = 'https://explorer.fuse.io'
```
Comment (sirpy) [inline src/server/utils/fuseapi.js]: not using the env variable for url

File: src/server/utils/fuseapi.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  runMethod(module, action, options) {
+    return new Promise((resolve, reject) => {
+      const queryParams = qs.stringify({
+        module,
+        action,
+        ...options
+      })
+      request(`${this.mainUrl}/api?${queryParams}`, { json: true }, (err, res, body) => {
```
Comment (sirpy) [inline src/server/utils/fuseapi.js]: use the frameworks we already use, such as crossfetch/axios, whatever we use in other places in the backend

File: src/server/verification/verificationAPI.js [hunk 4]
```diff
@@ -182,6 +184,29 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
...
+      // check if user send ether out of the good dollar system
+      let isUserSendEtherOutOfSystem = false
+      try {
+        const { result = [] } = await fuseapi.getTxList({
+          address: user.gdAddress,
+          page: 1,
+          offset: 0,
+          filterby: 'from'
+        })
+        isUserSendEtherOutOfSystem = result.some(r => Number(r.value) > 0)
```
Comment (sirpy) [inline src/server/verification/verificationAPI.js]: I feel this is too strict
maybe check if there are >1 such transactions so it looks like he is doing it repeatedly in a short time period

File: src/server/utils/fuseapi.js (hunk 3 above)
Comment (AlexeyKosinski) [reply src/server/utils/fuseapi.js]: we don't use crossfetch and axios yet. ok I will install axios

File: src/server/utils/fuseapi.js (hunk 2 above)
Comment (AlexeyKosinski) [reply src/server/utils/fuseapi.js]: done

File: src/server/server.config.js (hunk 1 above)
Comment (AlexeyKosinski) [reply src/server/server.config.js]: done

File: src/server/verification/verificationAPI.js (hunk 4 above)
Comment (AlexeyKosinski) [reply src/server/verification/verificationAPI.js]: but I think user mustn't use ether at all out of system.

File: src/server/utils/fuseapi.js (hunk 3 above)
Comment (AlexeyKosinski) [reply src/server/utils/fuseapi.js]: done

Comment (sirpy) [reply src/server/utils/fuseapi.js]: i requested whatever we use on the backend.

File: src/server/verification/verificationAPI.js [hunk 5]
```diff
@@ -190,7 +189,7 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
         const { result = [] } = await fuseapi.getTxList({
           address: user.gdAddress,
           page: 1,
-          offset: 0,
+          offset: 10,
```
Comment (sirpy) [inline src/server/verification/verificationAPI.js]: whats this?

File: src/server/utils/fuseapi.js (hunk 3 above)
Comment (AlexeyKosinski) [reply src/server/utils/fuseapi.js]: We don’t use any of frameworks to send requests in any places in the backend
that’s why I used request first time.
then replace with axios

Comment (AlexeyKosinski) [reply src/server/utils/fuseapi.js]: done

File: src/server/verification/verificationAPI.js (hunk 5 above)
Comment (AlexeyKosinski) [reply src/server/verification/verificationAPI.js]: it is strange. but offset in this api means "limit", amount of page. if we set 0 it returns all transactions. for ex. admin wallet returns 5400+ transactinos

Comment (AlexeyKosinski) [reply src/server/verification/verificationAPI.js]: when I set 10. it returns only 10 transactions

Comment (AlexeyKosinski) [reply src/server/verification/verificationAPI.js]: image

**Comment by sirpy** [comment]
@AlexeyKosinski build fails

**Comment by sirpy** [comment]
@AlexeyKosinski build + conflicts

**Comment by AlexeyKosinski** [comment]
> @AlexeyKosinski build + conflicts

done

**Pull request merged**